cf-promises/cf-promises.c

```diff
@@ -452,15 +452,25 @@ static void ShowVariablesFormatted(EvalContext *ctx)
         char *varname = VarRefToString(v->ref, true);
 
         Writer *w = StringWriter();
-        RvalWrite(w, v->rval);
+        bool json = false;
 
-        StringSet *tagset = EvalContextVariableTags(ctx, v->ref);
-        Buffer *tagbuf = StringSetToBuffer(tagset, ',');
+        switch (DataTypeToRvalType(v->type))
+        {
+        case RVAL_TYPE_CONTAINER:
+            json = true;
+            JsonWriteCompact(w, RvalContainerValue(v->rval));
+            break;
 
-        char *line;
-        const char *var_value;
+        default:
+            RvalWrite(w, v->rval);
+        }
 
-        if(StringIsPrintable(StringWriterData(w)))
+        const char *var_value;
+        if (json)
+        {
+            var_value = StringWriterData(w);
+        }
+        else if (StringIsPrintable(StringWriterData(w)))
         {
             var_value = StringWriterData(w);
         }
```

Comment:
Same action for both `if` branches, I would use an `||` for this test.

But I'm questioning whether we should short-circuit with `json` at all. It seems reasonable to me that the `StringIsPrintable` test should be applied to the JSON string as well, it's a string just like everything else. What do you think?